From 7fe5a8e874b2cae2b1ada3f6aab8ca6766753b1e Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Mon, 27 May 2019 09:36:28 +1000
Subject: [PATCH] Revert "mesa: unreference current winsys buffers when
 unbinding winsys buffers"

This reverts commit 12bf7cfecf52083c484602f971738475edfe497e.

This commits caused lots of problems:
https://bugs.freedesktop.org/show_bug.cgi?id=110721
https://bugs.freedesktop.org/show_bug.cgi?id=110761

Fixes: 12bf7cfecf52 ("mesa: unreference current winsys buffers when unbinding winsys buffers")
Pushing without review as we need to get it into next stable.
---
 src/mesa/main/context.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/mesa/main/context.c b/src/mesa/main/context.c
index 04ef4d500d8..34da16b88f4 100644
--- a/src/mesa/main/context.c
+++ b/src/mesa/main/context.c
@@ -1765,10 +1765,6 @@ _mesa_make_current( struct gl_context *newCtx,
 
          check_init_viewport(newCtx, drawBuffer->Width, drawBuffer->Height);
       }
-      else {
-         _mesa_reference_framebuffer(&newCtx->WinSysDrawBuffer, NULL);
-         _mesa_reference_framebuffer(&newCtx->WinSysReadBuffer, NULL);
-      }
 
       if (newCtx->FirstTimeCurrent) {
          handle_first_current(newCtx);
-- 
2.20.1

